src/app/(main)/product/[id]/page.tsx

```diff
@@ -0,0 +1,97 @@
+"use client";
+
+import React from "react";
+import { use } from "react";
+import { useProduct } from "@cps/trpc/use-product";
+import Head from "next/head";
+import Button from "@cps/components/button";
+import ImageGallery from "@cps/components/image/image-gallery"; 
+
+interface ProductDetailsPageProps {
+  params: Promise<{ id: string }>; 
+}
+
+const ProductDetailsPage: React.FC<ProductDetailsPageProps> = ({ params }) => {
+  const { id } = use(params); // Unwrap Promise with `use`
+  const { data, error, isLoading } = useProduct(id);
+
+  if (isLoading) {
+    return (
+      <div className="flex justify-center items-center h-screen">
+        <p className="text-gray-500">Loading product details...</p>
+      </div>
+    );
+  }
+
+  if (error || !data) {
+    return (
+      <div className="flex justify-center items-center h-screen">
+        <p className="text-red-500">Error loading product details. Please try again later.</p>
+      </div>
+    );
+  }
+
+  const { name, description, images, prices } = data;
+
+  // Transform images data to match ImageGallery's expected format
+  const galleryImages = images.map((image) => ({
+    src: image.url,
+    alt: image.title || "Product image",
+  }));
+
+  return (
+    <div>
+      {/* Head for SEO */}
+      <Head>
+        <title>{name} - Product Details</title>
+        <meta name="description" content={description || ""} />
+      </Head>
+
+      {/* Navigation Bar */}
+      <nav className="bg-gray-400 text-white p-4 flex justify-between">
+        <div className="absolute left-1/2 transform -translate-x-1/2">
+          <h1 className="text-lg font-bold">GPS Demo</h1>
+        </div>
+        <div className="flex items-center gap-4">
+          <a href="#" className="text-sm hover:underline">
+            Home
+          </a>
```

Comment:
This looks like a title in the navbar. Please remove it. In the #23, should already be implemented something for that.